### PR TITLE
fixed the multiple where-statements in the state affiliations query

### LINKED ISSUE #3238: [Bug] Revoking a user's access is causing them to disappear and not show in the Inactive tab
**Description-**

**Type of Bug-**

- [ ] Accessibility
- [x] Functional
- [ ] Usability
- [ ] API
- [x] Regression

**Steps to reproduce-**

1. Login with any account that has state admin access
2. Navigate to State Admin dashboard
3. Revoke a user from the "Active" tab
4. Switch to "Inactive" tab

**Role Affected-**

- [ ] All
- [ ] Federal Admin
- [x] State Admin
- [ ] State Staff
- [ ] State Contractor
- [ ] System Admin

**Environment-**

- [x] All
- [ ] PR #\_\_\_\_
- [ ] Staging
- [ ] Production

**Browser-**

- [ ] Firefox
- [ ] Chrome
- [ ] IE

**Expected results-**
* Revoked users should immediately appear in the inactive tab

**Actual results-**
* Revoked users are not appearing in the inactive tab when revoked

**Logs or error output/Screenshot/Video-** (at least 1 of these)
https://images.zenhubusercontent.com/113234315/ad82383b-d7f2-4a7f-865d-b42bd78c21cb/screen_recording_2021_07_01_at_12_32_27_pm.mov

---

Add additional labels (design, dev, compliance, BUG, etc) and size before submitting.

If the issue is needed to complete prioritized work for the CURRENT SPRINT, add it to the "This Sprint" pipeline. Otherwise, all other issues will be automatically added to the _unprioritized pipeline_ for prioritization in backlog refinement or sprint planning with Product.

### PR DESCRIPTION
**Pull request by thetif**
resolves #3238 

**Description-**
There was an issue with the sql query because it had a lot of complicated wheres. I just cleaned up that method. I also changed the arg coming into that query to be isFedAdmin instead of isAdmin to make it a little clearer.

**This pull request changes...**

- updated the where part of the state affiliations query
- changed an arg from isAdmin to isFedAdmin

**This pull request was tested in the follow ways…**

- list of backend cases
- manually tested all of the affiliation tabs as em@il.com and fedadmin to make sure that it's working as expected for State Admin and Federal Admin.

**Steps to manually verify this change...**

1. login as em@il.com
2. go to the state admin panel
3. under the pending tab should be Requested Role
4. under the inactive tab should be Revoked Role and Denied Role
5. under the active tab should be Regular User, State Admin, State Staff, and State Contractor
6. logout
7. login as fedadmin
8. go to the admin panel
3. under the pending tab should be Requested Role
4. under the inactive tab should be Revoked Role and Denied Role
5. under the active tab should be Regular User, Federal Admin, State Admin, State Staff, and State Contractor

### This pull request is ready to review when...

- [X] Automated tests are updated (and all tests are passing)
- [X] Changelog is updated as appropriate

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: api/db/affiliations.js [hunk 1]
```diff
@@ -14,57 +14,58 @@ const selectedColumns = [
...
-        state_id: stateId,
-        status: 'requested'
-      });
-  }
-  if (status === 'active') {
-    return query.where({
-        state_id: stateId,
-        status: 'approved'
+  if (affiliationStatuses) {
+    return query
```
Comment (tbolt) [inline api/db/affiliations.js:47]: clean! 🤩

**Review by tbolt — APPROVED** [review]
Lots of nice improvements. Good stuff.

Tested revoking multiple people from active and they appeared in the **Inactive** tab. As well I tested denying a request and that person appeared in the **Inactive** tab as well.

**Comment by tbolt** [comment]
Tested the following ways
| Given | When | Then | Result(s) |
------------ | ------------- | ------------ | -------------
| The em@il user | Viewing the state admin dashboard | I see the correct users under each tab | Pass ✔|
| The fedadmin user | Viewing the state admin dashboard | I see the correct users under each tab | Pass ✔ |
| The stateadmin user | Revoking a user from the active tab in the state admin dashboard | Switching to the inactive tab I see that revoked user | Pass ✔ |
| The stateadmin user | Restoring a user from the inactive tab in the state admin dashboard | switching to the active tab I see that restored user | Pass ✔|
| The stateadmin user | Approving a request in the state admin dashboard | shows the user under the active tab | Pass ✔|
| The stateadmin user | Denying a request in the state admin dashboard | shows the user under the inactive tab | Pass ✔|
| The stateadmin user | Reloading the page when on the Active tab | returns me to the same page on the Active tab | Fails. Not required behavior for this ticket afaik. Just documenting it. |

**Comment by akuas** [comment]
For em@il.com, there were no requested users or inactive users. There were active users with the same roles listed above.

**Review by akuas — APPROVED** [review]
All the state affiliations were correct for em@il.com and fedadmin.

**Review by jeromeleecms — APPROVED** [review]
Works as advertised as described in the test cases.

Couple notes: 

- Can't verify if this work with any other states other than AK at the moment since we don't have other users with different affiliations, but I don't see any hard coded values in the changed files.
- Fedadmin right now is limited to viewing AK only it seems, so I'll push this forward and restest what's going on once the Fedadmin dashboard is built.

**Pull request merged**